### PR TITLE
Check for duplicated Parameter/Buffer names in InferArguments (Issue #5292)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by dsharletg** [comment]
Also fixes #5160 ?

**Comment by steven-johnson** [comment]
> Also fixes #5160 ?

Unfortunately -- this only affects things that are visible as 'inputs' to a pipeline (included embedded Buffers); that bug happens for Func that are anywhere inside the pipeline.

**Comment by steven-johnson** [comment]
All green, PTAL

File: src/InferArguments.cpp [hunk 1]
```diff
@@ -32,24 +34,117 @@ class InferArguments : public IRGraphVisitor {
...
+            if (pob.param.is_buffer() && pob.param.buffer().same_as(b)) {
+                // Update this entry to have both the Parameter and Buffer.
+                pob.buffer = b;
+                return true;
+            } else {
+                // A Parameter and Buffer with the same name (but aren't connected) -> illegal
+                return dupe_names_error(name);
+            }
+        } else {
+            internal_error;  // There should be no empty ParamOrBuffers in the map.
```
Comment (abadams) [inline src/InferArguments.cpp:145]: Seems marginally better to make the comment a string argument to internal_error

**Review by abadams — APPROVED** [review]

**Pull request merged**